src/shared/components/header/header.component.js

```diff
@@ -36,9 +51,9 @@ export const Header = props => {
     <Container>
       <HeaderContainer>
         <Switch>
-          <Route path={ROUTES.benchmarkId} component={renderBenchmarkHeader} />
-          <Route path={ROUTES.home} component={renderHomeHeader} />
-          <Route exact path={ROUTES.base} component={renderHomeHeader} />
+          <Route path={`${ROUTES.benchmarkId}`} render={props => renderBenchmarkHeader(props)} />
```

Comment:
```suggestion
          <Route path={`${ROUTES.benchmarkId}`} render={renderBenchmarkHeader)} />
```
Staraj się nie definiować funkcji bezpośrednio w propie, jezeli nie jesteś do tego zmuszony.
W momencie, gdy robisz coś takiego, to komponent kiedy będzie miał okazję przerenderować się to na pewno to zrobi - ze względu na to, że przy takim zapisie za każdym razem tworzy się nowa funkcja. Dla reacta oznacza to, że prop się zmienił, więc czas przerenderować komponent ;)